src/Voron/Impl/FreeSpace/FreeSpaceHandling.cs

```diff
@@ -14,6 +15,18 @@ public sealed class FreeSpaceHandling : IFreeSpaceHandling
 
         private readonly FreeSpaceRecursiveCallGuard _guard;
 
+        private readonly Dictionary<long, SectionMetadata> _maxConsecutiveRangePerSection = new();
+
+        [StructLayout(LayoutKind.Sequential, Size = 8)]
+        public struct SectionMetadata()
+        {
+            public ushort Max = NumberOfPagesInSection;
+            public ushort StartBits = NumberOfPagesInSection;
+            public ushort EndBits = NumberOfPagesInSection;
```

Comment:
As I understand, `Max` is _total_ number of available pages in section. Is that correct? I have problem with understanding why both - `StartBits` and `EndBits` - are set to number of pages in section.

Reply:
The initial number is set to `Max` in order to sign that we have enough free space.
We update those numbers in `TryFindSmallValueMergingTwoSections`.

Reply:
We could use `ushort?` and initialize it to `null` but that wouldn't fit the struct size of 8.